pkg/highlight/highlighter.go

```diff
@@ -466,7 +467,16 @@ func (h *Highlighter) Highlight(input LineStates, startline, endline int) {
 		input.Unlock()
 	}
 
-	for i := startline; i <= endline; i++ {
+	isRehighlighting := false
+	searchNextSameState := false
+
+	input.Lock()
+	if endline < input.LinesNum()-1 {
+		isRehighlighting = true
+	}
+	input.Unlock()
```

Comment:
Looks overcomplicated and ad-hoc. And again, why do we need to explicitly distinguish between full highlighting and rehighlighting here?

Why not just something like:

```
	for i := startline; ; i++ {
		input.Lock()
		...
		if i >= endline {
			lastState = input.State(i) 
		}
		...
		input.Unlock()		

		if i >= endline && curState == lastState {
			break
		}
	}
```

No need for those ad-hoc extra variables then.

Reply:
> And again, why do we need to explicitly distinguish between full highlighting and rehighlighting here?

Now we don't since it's the exact same logic for both cases.
Thank you! I it didn't see it yesterday to shrink it down that much. :man_facepalming: